packages/editor/src/components/post-publish-panel/postpublish.js

```diff
@@ -41,142 +41,131 @@ const getFuturePostUrl = ( post ) => {
 	return post.permalink_template;
 };
 
-function CopyButton( { text, onCopy, children } ) {
-	const ref = useCopyToClipboard( text, onCopy );
-	return (
-		<Button __next40pxDefaultSize variant="secondary" ref={ ref }>
-			{ children }
-		</Button>
-	);
-}
```

Comment:
Extracting a component like this is a good idea, let's keep it! It's better than stuffing everything into one giant function.

The entire `showCopyConfirmation` logic including the timeouts should also be part of `CopyButton`.

Reply:
Good point!